src/cli_plugin/install/downloaders/http.js

```diff
@@ -2,11 +2,39 @@ import Wreck from 'wreck';
 import Progress from '../progress';
 import { fromNode as fn } from 'bluebird';
 import { createWriteStream } from 'fs';
+import HttpProxyAgent from 'http-proxy-agent';
+import URL from 'url';
+
+function getProxyAgent(sourceUrl) {
+  const httpProxy = process.env.HTTP_PROXY;
+  // we have a proxy detected, lets use it
+  if (httpProxy && httpProxy !== '') {
+    // get the hostname of the sourceUrl
+    const hostname = URL.parse(sourceUrl).hostname;
+    const noProxy = process.env.NO_PROXY || '';
+
+    // proxy if the hostname is not in noProxy
+    const shouldProxy = (noProxy.indexOf(hostname) === -1);
```

Comment:
We actually need to split the entries of `no_proxy` by comma before checking this, since there exist TLD domains, that are the prefix of another. So downloading a plugin from `foo.ac` would also not use the proxy if your env looks like `no_proxy="foo.academy,localhost"` even though it should be.

So we should split `no_proxy` by comma and check for an exact entry.

Also this way we don't support the more complex syntax of `no_proxy`, with ports and subdomain wildcards. We should think if these are important enough for the beginning that they must be there, or it's a nice to have to implement later.